src/torchrunx/environment.py

```diff
@@ -29,10 +35,40 @@ def slurm_workers() -> int:
     :rtype: int
     """
     # TODO: sanity check SLURM variables, commands
-    assert "SLURM_JOB_ID" in os.environ
+    assert in_slurm_job()
     if "SLURM_JOB_GPUS" in os.environ:
         # TODO: is it possible to allocate uneven GPUs across nodes?
         return len(os.environ["SLURM_JOB_GPUS"].split(","))
     else:
         # TODO: should we assume that we plan to do one worker per CPU?
         return int(os.environ["SLURM_CPUS_ON_NODE"])
+
+
+def auto_hosts() -> list[str]:
+    """
+    Automatically determine hostname list
+
+    :return: Hostnames in Slurm allocation, or ['localhost']
+    :rtype: list[str]
+    """
+    if in_slurm_job():
+        return ["localhost"]
+
+    return slurm_hosts()
+
+
+def auto_workers() -> int:
+    """
+    Automatically determine number of workers per host
+
+    :return: Workers per host
+    :rtype: int
+    """
+
+    if in_slurm_job:
+        _cpus = os.cpu_count()
+        cpus = 1 if _cpus is None else _cpus
+        gpus = torch.cuda.device_count()
+        return cpus if gpus == 0 else gpus
```

Comment:
I think you could simply write this as `return torch.cuda.device_count() or os.cpu_count() or 1`